middleware/validator.ts

```diff
@@ -0,0 +1,13 @@
+import { NextFunction, Request, Response } from "express";
+import { AnyObjectSchema } from "yup";
+
+export const validateMiddleware = (schema: AnyObjectSchema) => {
```

Comment:
I suggest changing the name of the function *validateMiddleware* to just *validate* as it will be more readable in routes. eg:
```
router.route("/api/auth/register")
    .post(
        validate(registrationSchema),
        register(...)
    )
```